public/app/core/components/NavBar/navBarItem-translations.ts

```diff
@@ -1,66 +1,100 @@
-  dashboards: defineMessage({ id: 'nav.dashboards', message: 'Dashboards' }),
-  'dashboards/browse': defineMessage({ id: 'nav.manage-dashboards', message: 'Browse' }),
-  'dashboards/playlists': defineMessage({ id: 'nav.playlists', message: 'Playlists' }),
-  'dashboards/snapshots': defineMessage({ id: 'nav.snapshots', message: 'Snapshots' }),
-  'dashboards/library-panels': defineMessage({ id: 'nav.library-panels', message: 'Library panels' }),
-  'dashboards/new': defineMessage({ id: 'nav.new-dashboard', message: 'New dashboard' }),
-  'dashboards/folder/new': defineMessage({ id: 'nav.new-folder', message: 'New folder' }),
-
-  explore: defineMessage({ id: 'nav.explore', message: 'Explore' }),
-
-  alerting: defineMessage({ id: 'nav.alerting', message: 'Alerting' }),
-  'alerting-legacy': defineMessage({ id: 'nav.alerting-legacy', message: 'Alerting (legacy)' }),
-  'alert-list': defineMessage({ id: 'nav.alerting-list', message: 'Alert rules' }),
-  receivers: defineMessage({ id: 'nav.alerting-receivers', message: 'Contact points' }),
-  'am-routes': defineMessage({ id: 'nav.alerting-am-routes', message: 'Notification policies' }),
-  channels: defineMessage({ id: 'nav.alerting-channels', message: 'Notification channels' }),
-
-  silences: defineMessage({ id: 'nav.alerting-silences', message: 'Silences' }),
-  groups: defineMessage({ id: 'nav.alerting-groups', message: 'Groups' }),
-  'alerting-admin': defineMessage({ id: 'nav.alerting-admin', message: 'Admin' }),
-
-  cfg: defineMessage({ id: 'nav.config', message: 'Configuration' }),
-  datasources: defineMessage({ id: 'nav.datasources', message: 'Data sources' }),
-  correlations: defineMessage({ id: 'nav.correlations', message: 'Correlations' }),
-  users: defineMessage({ id: 'nav.users', message: 'Users' }),
-  teams: defineMessage({ id: 'nav.teams', message: 'Teams' }),
-  plugins: defineMessage({ id: 'nav.plugins', message: 'Plugins' }),
-  'org-settings': defineMessage({ id: 'nav.org-settings', message: 'Preferences' }),
-  apikeys: defineMessage({ id: 'nav.api-keys', message: 'API keys' }),
-  serviceaccounts: defineMessage({ id: 'nav.service-accounts', message: 'Service accounts' }),
-
-  live: defineMessage({ id: 'nav.live', message: 'Event streaming' }),
-  'live-status': defineMessage({ id: 'nav.live-status', message: 'Status' }),
-  'live-pipeline': defineMessage({ id: 'nav.live-pipeline', message: 'Pipeline' }),
-  'live-cloud': defineMessage({ id: 'nav.live-cloud', message: 'Cloud' }),
-
-  help: defineMessage({ id: 'nav.help', message: 'Help' }),
-
-  'profile-settings': defineMessage({ id: 'nav.profile/settings', message: 'Preferences' }),
-  'change-password': defineMessage({ id: 'nav.profile/password', message: 'Change password' }),
-  'sign-out': defineMessage({ id: 'nav.sign-out', message: 'Sign out' }),
-};
-
-export default TRANSLATED_MENU_ITEMS;
+export default function getNavTranslation(navId: string | undefined) {
```

Comment:
Probably my suggestion is overkill.

I tried adding a new `case` with the same value inside the `switch` statement and JS/TS didn't give any warning about that. Maybe it could be fixed if we move it to a `JSON`(something like the snippet below). I tried it and it gives an early warning to a developer in case they provided an existing key in the object. One bad thing is that we created an unnecessary JSON 😄 .

```
const TRANSLATED_MENU_ITEMS: { [key: string ]: { key: string; defaultValue: string}} = {
  "home": { key: 'nav.home', defaultValue: 'Home'},  
}

export default function getNavTranslation(navId: string) {
  const {key, defaultValue} = TRANSLATED_MENU_ITEMS[navId]

  return t(key, defaultValue);
}

```

Reply:
Aha! got it. I think yours is better(please disregard my previous comment  I just noticed that duplicated case can be solve by `no-duplicate-case` rule)